### PR TITLE
Made clip plane settings for DrawState

### PR DESCRIPTION
**Pull request by davidpiuva**
Allows changing clip planes for special cases.

### CODE REVIEW

**Comment by davidpiuva** [comment]
@christopherlagerhult Peer review?

File: source/draw/DrawState.ooc [hunk 1]
```diff
@@ -38,6 +38,8 @@ DrawState: cover {
 	_sourceNormalized := FloatBox2D new(0.0f, 0.0f, 1.0f, 1.0f)
 	_focalLengthNormalized := 0.0f // Relative to image width
 	targetTransform := FloatTransform2D identity // 2D scaling done after projection matrix but before w division
+	nearPlane: Float = 0.01f
+	farPlane: Float = 12500.0f
```
Comment (christopherlagerhult) [inline source/draw/DrawState.ooc]: Just a though. Do you really need to initialize these variables if you're going to have them defaulted in `_createProjection` with the same value?

**Comment by christopherlagerhult** [comment]
Other then the comment, I'm happy

File: source/draw/DrawState.ooc (hunk 1 above)
Comment (davidpiuva) [inline source/draw/DrawState.ooc]: If not set, the uninitialized data will be given to `_createProjection` and override the default values with uninitialized values (possibly NaN causing a once in a million crash).

Comment (christopherlagerhult) [inline source/draw/DrawState.ooc]: Yeah but I was thinking more in term of that `_createProjection` could be called w/o those parameters then. Never mind though, just a thought.

Comment (marcusnaslund) [inline source/draw/DrawState.ooc]: But maybe you want to call `_createProjection` with custom argument values?

File: source/draw/gpu/opengl/OpenGLPacked.ooc [hunk 2]
```diff
@@ -115,14 +115,12 @@ OpenGLPacked: abstract class extends GpuImage {
 	_createView: func (targetSize: FloatVector2D, normalizedView: FloatTransform3D) -> FloatTransform3D {
 		this _toLocal * normalizedView normalizedToReference(targetSize) * this _toLocal
 	}
-	_createProjection: func (targetSize: FloatVector2D, focalLengthPerWidth: Float) -> FloatTransform3D {
+	_createProjection: func (targetSize: FloatVector2D, focalLengthPerWidth, nearPlane, farPlane: Float) -> FloatTransform3D {
```
Comment (davidpiuva) [inline source/draw/gpu/opengl/OpenGLPacked.ooc]: Updated

**Pull request merged**